tests/cephfs/cephfs_mirroring/test_cephfs_mirror_auth_caps.py

```diff
@@ -0,0 +1,522 @@
+                )
+                log.info(
+                    f"Snaps Synced: {result_snap_f1['snaps_synced']} of '{result_snap_f1['snapshot_name']}'"
+                )
+
+                log.info("All snapshots synced.")
+            else:
+                log.error("One or both snapshots not found or not synced.")
+
+            log.info("Validate the snapshots and data synced to the target cluster")
+            target_mount_path1 = f"/mnt/remote_dir1_{caps}"
+            target_mount_path2 = f"/mnt/remote_dir2_{caps}"
+            target_client_user = "client.admin"
+            source_path1 = subvol1_path
+            source_path2 = subvol2_path
+            snapshot_name1 = snap1
+            snapshot_name2 = snap2
+            expected_files1 = file_name1
+            expected_files2 = file_name2
+
+            (
+                success1,
+                result1,
+            ) = fs_mirroring_utils.list_and_verify_remote_snapshots_and_data(
+                target_clients[0],
+                target_mount_path1,
+                target_client_user,
+                source_path1,
+                snapshot_name1,
+                expected_files1,
+            )
+            (
+                success2,
+                result2,
+            ) = fs_mirroring_utils.list_and_verify_remote_snapshots_and_data(
+                target_clients[0],
+                target_mount_path2,
+                target_client_user,
+                source_path2,
+                snapshot_name2,
+                expected_files2,
+            )
+
+            if success1 and success2:
+                log.info(
+                    "Test Completed Successfully.All snapshots and data are synced to target cluster"
+                )
+            else:
+                log.info(
+                    "Test Failed. Snapshots or Data are missing on target cluster."
+                )
+            log.info(f"Status of {snap1}: {result1}")
+            log.info(f"Status of {snap2}: {result2}")
+            source_clients[0].exec_command(
+                sudo=True,
+                cmd=f"rm -rf {kernel_mounting_dir_1}{subvol1_path}hello_kernel_{caps}",
+            )
+            source_clients[0].exec_command(
+                sudo=True,
+                cmd=f"rm -rf {fuse_mounting_dir_1}{subvol2_path}hello_fuse_{caps}",
+            )
+            source_clients[0].exec_command(
+                sudo=True,
+                cmd=f"ceph fs subvolume snapshot rm {target_fs} subvol_caps_1 {snap1} --group_name subvolgroup_1",
+            )
+
+            source_clients[0].exec_command(
+                sudo=True,
+                cmd=f"ceph fs subvolume snapshot rm {target_fs} subvol_caps_2 {snap2} --group_name subvolgroup_1",
+            )
+        return 0
+    except Exception as e:
+        log.error(e)
+        log.error(traceback.format_exc())
+        return 1
+    finally:
+        log.info("Clean up the system")
```

Comment:
Can we have a generic method for all mandatory cleanup events in case of mirroring tests? I see the same steps repeated in every other test of mirroring.

Reply:
Same goes as below comment

Reply:
> Can we have a generic method for all mandatory cleanup events in case of mirroring tests? I see the same steps repeated in every other test of mirroring.

+1